platform/ios/src/MGLMapView.h

```diff
@@ -1300,6 +1300,34 @@ IB_DESIGNABLE
  */
 - (NS_ARRAY_OF(id <MGLFeature>) *)visibleFeaturesInRect:(CGRect)rect inStyleLayersWithIdentifiers:(nullable NS_SET_OF(NSString *) *)styleLayerIdentifiers predicate:(nullable NSPredicate *)predicate NS_SWIFT_NAME(visibleFeatures(in:styleLayerIdentifiers:predicate:));
 
+/**
+ Returns an array of map features for the given source, restricted to the given 
+ source layer and filtered by the given predicate.
+ 
+ Each object in the returned array represents a feature for the
+ current style and provides access to attributes specified by the relevant
+ <a href="https://www.mapbox.com/mapbox-gl-style-spec/#sources">tile sources</a>.
+ The returned array includes features specified in vector and GeoJSON tile
+ sources but does not include anything from raster, image, or video sources.
+
+ Features come from tiled vector data or GeoJSON data that is converted to tiles
+ internally, so feature geometries are clipped at tile boundaries and features
+ may appear duplicated across tiles. For example, suppose the specified
+ rectangle intersects with a road that spans the screen. The resulting array
+ includes those parts of the road that lie within the map tiles covering the
+ specified rectangle, even if the road extends into other tiles. The portion of
+ the road within each map tile is included individually.
+ 
+ @param sourceIdentifier The source to query.
+ @param sourceLayerIdentifier The source layer to include in the query. Only the 
+    features contained in this source layer are included in the returned array. 
+    The source layer is ignored for GeoJSON sources and required for vector sources.
+ @param predicate A predicate to filter the returned features.
+ @return An array of objects conforming to the `MGLFeature` protocol that
+    represent features in the sources used by the current style.
+ */
+- (NS_ARRAY_OF(id <MGLFeature>) *)featuresWithSourceIdentifier:(NSString *)sourceIdentifier sourceLayerIdentifier:(nullable NSString *)sourceLayerIdentifier predicate:(nullable NSPredicate *)predicate NS_SWIFT_NAME(features(sourceIdentifier:sourceLayerIdentifier:predicate:));
```

Comment:
How about `-featuresLoadedInSource:sourceLayerIdentifier:predicate:`?

Especially if this method only works with a single source at a time, I think it would be desirable to require the developer to pass in an MGLSource object instead of a source identifier. That encourages the developer to obtain an existing source and helps to clarify the distinction between a source identifier and a source layer identifier.

Additionally, we probably want to emphasize that this method returns the features that have already loaded, not all the possible features in the source (which for a vector source could be quite unwieldy). This isn't quite a reverse geocoding API, after all.